datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVDataMapProvider.scala

```diff
@@ -95,8 +95,11 @@ class MVDataMapProvider(
       dataMapSchema.getRelationIdentifier.getTableName,
       true)
     dropTableCommand.processMetadata(sparkSession)
-    DataMapStoreManager.getInstance.unRegisterDataMapCatalog(dataMapSchema)
+    // First, drop datamapschema and unregister datamap from catalog, because if in
+    // case, unregister fails, datamapschema will not be deleted from system and cannot
+    // create datamap also again
     DataMapStoreManager.getInstance().dropDataMapSchema(dataMapSchema.getDataMapName)
+    DataMapStoreManager.getInstance.unRegisterDataMapCatalog(dataMapSchema)
```

Comment:
Now If dropDataMapSchema throws exception, unregister will not happen. 
better to keep dropDataMapSchema in try catch and unRegisterDataMapCatalog in finally block

Reply:
I think if drop datamap schema fails let it fail , no need to unregister, because it can valid reason, if we unregister even if it fails later that schema file might create problem . @indhumati please check code once and confirm

Reply:
@akashrn5 dropDatamapschema can fail if only dataschema file is not present physically. May be deleted in backend. In that case also, we need to do unregister, to allow user to create datamap again